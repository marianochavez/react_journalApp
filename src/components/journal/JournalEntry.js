import React from "react";

export const JournalEntry = () => {
  return (
    <div className="journal__entry pointer">
      <div
        className="journal__entry-picture"
        style={{
          backgroundSize: "cover",
          backgroundImage:
            "url(https://www.dzoom.org.es/wp-content/uploads/2017/07/seebensee-2384369-810x540.jpg)",
        }}
      ></div>

      <div className="journal__entry-body">
        <p className="journal__entry-title"> Un nuevo d√≠a</p>
        <p className="journal__entry-content">
          Occaecat non aute nulla excepteur sunt cupidatat eu.
        </p>
      </div>

      <div className="journal__entry-date-box">
        <span>Monday</span>
        <h4>28</h4>
      </div>
    </div>
  );
};
