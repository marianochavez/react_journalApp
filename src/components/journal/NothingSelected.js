import React from 'react';

export const NothingSelected = () => {
  return <div className='nothing__main-content'>
      <p>
          Select something
            <br/>
            or create an entry!
      </p>

      <i class="fa-brands fa-react fa-4x"></i>
  </div>;
};
