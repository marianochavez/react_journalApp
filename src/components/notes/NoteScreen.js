import React from "react";
import { NotesAppBar } from "./NotesAppBar";

export const NoteScreen = () => {
  return (
    <div className="notes__main-content">
      <NotesAppBar />

      <div className="notes__content">
        <input
          type="text"
          placeholder="Some awesome title"
          className="notes__title-input"
        />

        <textarea
          placeholder="What happened today"
          className="notes__textarea"
        ></textarea>

        <div className="notes__image">
          <img
            src="https://cdn.pixabay.com/photo/2018/01/14/23/12/nature-3082832__340.jpg"
            alt="note"
          />
        </div>
      </div>
    </div>
  );
};
