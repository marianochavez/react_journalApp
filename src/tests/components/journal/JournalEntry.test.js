import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { mount } from "enzyme";
import { Provider } from "react-redux";

import { JournalEntry } from "../../../components/journal/JournalEntry";
import { activeNote } from "../../../actions/notes";


const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};

let store = mockStore(initState);
store.dispatch = jest.fn();

const note = {
    id: 10,
    date: 213,
    title: "Some awesome title",
    body: "What happened today",
    url: 'https://www.google.com'
};

const wrapper = mount(
  <Provider store={store}>
    <JournalEntry {...note}/>
  </Provider>
);

describe('JournalEntry tests', () => { 
    
    test('should show correctly', () => { 
        expect(wrapper).toMatchSnapshot();
     });

     test('should active note', () => { 
         wrapper.find('.journal__entry').prop('onClick')();
         expect(store.dispatch).toHaveBeenCalledWith(
             activeNote(note.id, {...note})
         )
      })
 })