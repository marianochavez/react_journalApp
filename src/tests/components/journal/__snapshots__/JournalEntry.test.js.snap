// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JournalEntry tests should show correctly 1`] = `
<div
  className="journal__entry pointer animate__animated animate__backInUp"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(https://www.google.com)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      Some awesome title
    </p>
    <p
      className="journal__entry-content"
    >
      What happened today
    </p>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
      Wednesday
    </span>
    <h4>
      31st
    </h4>
  </div>
</div>
`;
